discovery/manager_test.go

```diff
@@ -1115,7 +1115,11 @@ func (tp mockdiscoveryProvider) Run(ctx context.Context, upCh chan<- []*targetgr
 		for i := range u.targetGroups {
 			tgs[i] = &u.targetGroups[i]
 		}
-		upCh <- tgs
+		select {
+		case <-ctx.Done():
+			return
+		case upCh <- tgs:
+		}
 	}
 	<-ctx.Done()
```

Comment:
I think this `<-ctx.Done()` could be removed now.

Reply:
Oh, sorry for my delayed response!
I added the `case <-ctx.Done():`, so if the parent goroutine is killed by `t.Fatal()` at line 681, the child goroutine will also exit by choosing this case.